js/date-filter.js

```diff
@@ -125,6 +146,127 @@ DateFilter.prototype.handleModifyEvent = function(e, opts) {
+  // set the selected date range in the grid
+  this.handleDateGridRange($dateBegin, $dateEnd);
+};
+
+DateFilter.prototype.handleDateGridRange = function($dateHoverBegin, $dateHoverEnd) {
+  this.$grid.find('li').removeClass();
+
+  $dateHoverBegin.addClass('selected begin');
+  $dateHoverEnd.addClass('selected end');
+
+  if (!$dateHoverBegin.is($dateHoverEnd)) {
+    $dateHoverBegin.nextUntil('.end').addClass('selected');
+    $dateHoverEnd.prevUntil('.begin').addClass('selected');
+  }
+};
+
+DateFilter.prototype.handleMinDateSelect = function() {
+  var self = this;
+  var $dateBegin = this.$grid.find('.begin');
+  var $dateEnd = this.$grid.find('.end');
+
+  this.$grid.show().removeClass('pick-max').addClass('pick-min');
+  this.$grid.find('.active').removeClass('active');
+  $dateBegin.addClass('active');
+
+  this.$grid.find('li').hover(
+    function() {
+      var dateBeginNum = parseInt($(this).parent().attr('data-year') + $(this).attr('data-month'));
+      var dateEndNum = parseInt($dateEnd.parent().attr('data-year') + $dateEnd.attr('data-month'));
+
+      if (dateBeginNum <= dateEndNum) {
+        self.$grid.removeClass('invalid');
+        self.handleDateGridRange($(this), $dateEnd, $dateBegin, $dateEnd);
+      }
+      else {
+        self.$grid.addClass('invalid');
+      }
+    },
+    function() {
+      self.handleDateGridRange($dateBegin, $dateEnd);
+      $dateBegin.addClass('active');
+    }
+  );
+};
+
+DateFilter.prototype.handleMaxDateSelect = function() {
+  var self = this;
+  var $dateBegin = this.$grid.find('.begin');
+  var $dateEnd = this.$grid.find('.end');
+
+  this.$grid.show().removeClass('pick-min').addClass('pick-max');
+  this.$grid.find('.active').removeClass('active');
+  $dateEnd.addClass('active');
+
+  this.$grid.find('li').hover(
+    function() {
+      var dateBeginNum = parseInt($dateBegin.parent().attr('data-year') + $dateBegin.attr('data-month'));
+      var dateEndNum = parseInt($(this).parent().attr('data-year') + $(this).attr('data-month'));
+
+      if (dateBeginNum <= dateEndNum) {
+        self.$grid.removeClass('invalid');
+        self.handleDateGridRange($dateBegin, $(this), $dateBegin, $dateEnd);
+      }
+      else {
+        self.$grid.addClass('invalid');
+      }
+    },
+    function() {
+      self.handleDateGridRange($dateBegin, $dateEnd);
+      $dateEnd.addClass('active');
+    }
+  );
+};
+
+DateFilter.prototype.dateGridSelect = function (e) {
+  var value = [];
+  var $selectDate = $(e.target).parent();
+
+  if (this.$grid.hasClass('pick-min')) {
+    value[0] = $selectDate.data('month') + '/01/' + $selectDate.parent().attr('data-year');
+    value[1] = this.$maxDate.val();
+  }
+  else {
+    value[0] = this.$minDate.val();
+    value[1] = $selectDate.data('month') + '/01/' + $selectDate.parent().attr('data-year');
+  }
+
+  if (!this.$grid.hasClass('invalid')) {
+    this.$grid.removeClass('pick-min pick-max');
+    this.$grid.find('li').unbind('mouseenter mouseleave');
```

Comment:
Yeah, pretty much unbinding hover so it doesn't keep hover date changing after date select.